### PR TITLE
[Enhancement] nvm segment now shows version when 'nvm current' is 'system'

### PR DESCRIPTION
**Pull request by Glutnix**
#### Description
This improves the `nvm` segment, and has it render the version of system node when `nvm current` outputs `system`. It also prevents truncating the first character off `system`

screen shot 2018-12-03 at 4 55 08 pm

#### Questions
I haven't made a test for this enhancement, as I can't find a reliable way to fake the version of system.

I don't think I've put my patch on the right branch... which branch should have I branched from? I think I branched from `next` ... :/

### CODE REVIEW

File: powerlevel9k.zsh-theme [hunk 1]
```diff
@@ -1228,11 +1228,18 @@ prompt_nvm() {
...
   [[ -z "${node_version}" || ${node_version} == "none" ]] && return
+  if [[ "$node_version" == "system" ]]; then
+    node_version="$($(nvm which system) --version) system"
+  fi
 
   nvm_default=$(nvm_version default)
   [[ "$node_version" =~ "$nvm_default" ]] && return
 
-  $1_prompt_segment "$0" "$2" "magenta" "black" "${node_version:1}" 'NODE_ICON'
+  if [[ "${node_version:0:1}" == "v" ]]; then
```
Comment (dritter) [inline powerlevel9k.zsh-theme]: This just looks at the returned version and tries to get rid of the `v` at the beginning, right? Wouldn't it be more clear to parse it with a regular expression?
```zsh
[[ "${node_version}" =~ "^v([0-9.]+)" ]] && node_version="${match[1]}"
```

**Comment by dritter** [comment]
Hi @Glutnix ,
Thanks for your PR. 👍 

To answer your questions:

You are already on `next`, but you need to update your PR. I did some changes/merged some features since you created your fork.

To create a mock that depends on a given argument, have a look at the [go tests](https://github.com/bhilburn/powerlevel9k/blob/ad9321795b18f1c578e3f3a9072aa1520dc2db75/test/segments/go_version.spec#L15-L24).

**Comment by Glutnix** [comment]
Hi @dritter, thanks for the friendly welcome, and the very useful advice on the tests.

I've improved my patch on this PR on the `next` branch, along with adding a test, which is passing!

**Comment by dritter** [comment]
Thx @Glutnix and welcome to P9K!

**Comment by Glutnix** [comment]
🙌
Thanks for accepting my PR :)

**Pull request merged**